driver/labjack/reader_source.cpp

```diff
@@ -13,6 +13,37 @@
 ///////////////////////////////////////////////////////////////////////////////////
 //                                   ReaderSource                                //
 ///////////////////////////////////////////////////////////////////////////////////
+labjack::ReaderSource::ReaderSource(
+        const std::shared_ptr<task::Context> &ctx,
+        const synnax::Task task,
+        const ReaderConfig &reader_config
+) : ctx(ctx),
+    task(task),
+    reader_config(reader_config) {
+    auto breaker_config = breaker::Config{
+            .name = task.name,
+            .base_interval = 1 * SECOND,
+            .max_retries = 20,
+            .scale = 1.2,
+    };
+    this->breaker = breaker::Breaker(breaker_config);
+    this->open_device();
+}
+
+void labjack::ReaderSource::open_device(){
+    std::lock_guard<std::mutex> lock(labjack::device_mutex);
```

Comment:
This function is very similar to the one in the scanner. Why not consolidate a single function that accepts a serial number and adds it to the map, and then call that function from both the scanner and reader source?

Also, what about the writer? Does that need to be opened?

Reply:
I also don't think we should use a global map here. Why not build a class called "DeviceManager" that is a field on the factory, and that you pass into the constructor of both the scanner and reader tasks. Dependency injection is much clearer than using globals

Reply:
i was lazy

Reply:
i created device manager class